### DISCUSSION TITLE
docker.io auth with a token - configuration confirmation

### DISCUSSION THREAD

**Discussion by lmilbaum** [renovatebot/renovate#23534] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

latest

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I'd like renovate to pull docker images from docker.io with authentication to avoid the pull rate limit issues

### What you tried so far.

The following CI/CD variables defined:
DOCKER_AUTH_USERNAME
DOCKER_AUTH_TOKEN

Is that the correct configuration?
```
  "hostRules": [
    {
        "hostType": "docker",
        "matchHost": "docker.io",
        "username": process.env.DOCKER_AUTH_USERNAME,
        "password": process.env.DOCKER_AUTH_TOKEN
    }
  ]
```

Can I add it to the default.json file instead of config.js?

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by viceice** [comment, marked as answer]
no, there is no environment replace inside JSON file
